database.py

```diff
@@ -1,39 +1,79 @@
+import peewee
+
 class Result:
     def __init__(self, is_success=True, data=None):
         self.is_success = is_success
         self.data = data
 
 class Database:
     def __init__(self, db_name, uname="test", pwd="test", host="localhost"):
-        self.db_name = db_name
-
-    def start_connection(self):
-        return Result(True)
+        self.db = peewee.PostgresqlDatabase(
+            db_name,
+            user=uname,
+            password=pwd,
+            host=host
+        )
 
-    def end_connection(self):
-        return Result(True)
+        # This is taken mostly from the Peewee sample app
+        class BaseModel(peewee.Model):
+            class Meta:
+                database = self.db
+        
+        class Resume(BaseModel):
+            user_id = peewee.IntegerField(primary_key=True)
+            resume = peewee.TextField()
+            id = peewee.AutoField()
 
-    def add_admin(self, user_id):
-        return Result(True)
+        self.Resume = Resume
 
-    def remove_admin(self, user_id):
-        return Result(True)
+    def start_connection(self):
+        self.db.connect()
+        self.db.create_tables([self.Resume])
+        self.db.commit()
 
-    def is_admin(self, user_id):
-        return Result(True)
+    def end_connection(self):
+        self.db.close()
 
     def push_resume(self, user_id, resume):
-        return Result(True)
+        try:
+            q = self.Resume(user_id=user_id, resume=resume)
+            n_rows_modified = q.save()
+            return Result(is_success=n_rows_modified == 1)
+        except peewee.IntegrityError:
+            self.db.rollback()
+            return Result(is_success=False)
     
     def pop_resume(self):
-        return Result(True)
+        try:
+            r = self.Resume.select(self.Resume.user_id, self.Resume.resume).order_by(self.Resume.id).get()
+            user_id = r.user_id
+            resume = r.resume
+            r.delete_instance()
+            return Result(is_success=True, data=(user_id, resume))
+        except peewee.DoesNotExist:
+            return Result(is_success=False)
 
     def delete_resume(self, user_id):
```

Comment:
Are delete and pop doing the same thing with just the id?

Reply:
So maybe just have pop do what it's doing then have delete return pop but only return the resume?